app/src/main/java/com/bashkevich/androidfundamentals/model/MoviesDao.kt

```diff
@@ -0,0 +1,29 @@
+package com.bashkevich.androidfundamentals.model
+
+import androidx.lifecycle.LiveData
+import androidx.room.*
+import com.bashkevich.androidfundamentals.model.entity.ActorEntity
+import com.bashkevich.androidfundamentals.model.entity.MovieActorCrossRef
+import com.bashkevich.androidfundamentals.model.entity.MovieEntity
+import com.bashkevich.androidfundamentals.model.entity.MovieWithActors
+
+@Dao
+interface MoviesDao {
+
+    @Transaction
+    @Query("SELECT * FROM movies where movie_id=:movieId")
+    suspend fun getMovieWithActorsById(movieId: Int): MovieWithActors
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertMovie(movie: MovieEntity)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertAllActors(actors: List<ActorEntity>)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertJoin(join: MovieActorCrossRef)
+
+    @Query("SELECT * FROM movies")
+    fun getAllMovies(): LiveData<List<MovieEntity>>
```

Comment:
Why other methods suspend and this one is using LiveData?

Reply:
because I need for updating list of movies

Reply:
Rename method that it reflects also subscription?

Reply:
why rename method?